### ISSUE TITLE
Julia SDK should be automatically refreshing the jwt tokens if they're expired

### ISSUE THREAD

**Issue by NHDaly** [RelationalAI/rai-sdk-julia#30]
```
“error validating jwt token: error validating jwt token: square/go-jose/jwt: validation failed, token is expired (exp): error validating claims”.
```

If you don't use the julia SDK for a short time, the tokens expire, and queries fail. We should be auto-refreshing the tokens.

**Comment by NHDaly** [comment]
SDK should check the expiration time before including a token in a request, and if it's expired, it should request a new one.

**Comment by NHDaly** [comment]
@NRHelmi i think you are working on a PR to address this right now?

**Comment by NHDaly** [comment]
Closed by #31.
